lib/src/client/http2_connection.dart

```diff
@@ -69,25 +73,36 @@ class Http2ClientConnection implements connection.ClientConnection {
 
   ConnectionState get state => _state;
 
+  static const _estimatedRoundTripTime = const Duration(milliseconds: 20);
+
   Future<ClientTransportConnection> connectTransport() async {
-    var socket = await Socket.connect(host, port);
-    if (_state == ConnectionState.shutdown) {
-      socket.destroy();
-      throw _ShutdownException();
-    }
     final securityContext = credentials.securityContext;
-    if (securityContext != null) {
-      socket = await SecureSocket.secure(socket,
-          host: authority,
+    Socket socket;
+    if (securityContext == null) {
+      socket = await Socket.connect(host, port);
+    } else {
+      final socket = await SecureSocket.connect(host, port,
```

Comment:
You're creating a new local variable here, so the outer `socket` remains null.  Perhaps name this `secureSocket`, and then set `socket = secureSocket` after the "h2" if check below?

Reply:
(also, is there any way to test this code path?)

Reply:
Oops
Thanks for catching this!